posthog/hogql/property.py

```diff
@@ -28,11 +28,11 @@ def __init__(self):
         super().__init__()
         self.has_aggregation = False
 
-    def visit(self, node):
+    def visit(self, node: ast.AST, tag: Optional[str] = None):
```

Comment:
So `tag` is clunky and ambiguous. I think the real issue here is that the AST has no concept of a column , which is key, as a column is a very special of an expression (`columnExpr` in the syntax). The rule is basically: `ast.Alias` is allowed as an `ast.Column` expr, and nowhere else. So if we have `ast.Column`, we'll be able to check whether the parent is a column simply using `stack`.